### PR TITLE
Fhac 600

### PR DESCRIPTION
**Pull request by TabassumJafri**
1. Added the ATNA logging for PD deferred request.

### CODE REVIEW

**Comment by sailajaa** [comment]
Reviewed Regression script changes. I am good with script updates for PD deferred service and properties file.

File: Product/Production/Gateway/PatientDiscovery_10/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/_10/gateway/ws/EntityPatientDiscoveryDeferredRequestSecured.java [hunk 1]
```diff
@@ -57,13 +57,14 @@ public EntityPatientDiscoveryDeferredRequestSecured() {
...
+        afterReturningBuilder = MCCIIN000002UV01EventDescriptionBuilder.class,
+        serviceType = "Patient Discovery Deferred Request", version = "1.0")
+    @Override
     public MCCIIN000002UV01 processPatientDiscoveryAsyncReq(RespondingGatewayPRPAIN201305UV02SecuredRequestType request) {
         AssertionType assertion = getAssertion(context, null);
 
         return outboundPatientDiscoveryRequest.processPatientDiscoveryAsyncReq(request.getPRPAIN201305UV02(),
-                assertion, request.getNhinTargetCommunities());
+            assertion, request.getNhinTargetCommunities(), getWebContextProperties(context));
     }
```
Comment (alameluchidambaram) [inline Product/Production/Gateway/PatientDiscovery_10/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/_10/gateway/ws/EntityPatientDiscoveryDeferredRequestSecured.java:68]: Can you explain why does Entity Outbound requires WebContextProperties for PatientDiscoveryDeferredRequest service?

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/deferred/request/proxy/EntityPatientDiscoveryDeferredRequestProxy.java [hunk 2]
```diff
@@ -28,11 +28,13 @@
...
 
 import org.hl7.v3.MCCIIN000002UV01;
 import org.hl7.v3.PRPAIN201305UV02;
 
 public interface EntityPatientDiscoveryDeferredRequestProxy {
+
     public MCCIIN000002UV01 processPatientDiscoveryAsyncReq(PRPAIN201305UV02 request, AssertionType assertion,
-            NhinTargetCommunitiesType target);
+        NhinTargetCommunitiesType target, Properties webContextProperties);
 }
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/deferred/request/proxy/EntityPatientDiscoveryDeferredRequestProxy.java:39]: same question as above.

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/deferred/request/proxy/EntityPatientDiscoveryDeferredRequestProxyWebServiceSecuredImpl.java [hunk 3]
```diff
@@ -55,23 +56,26 @@ protected WebServiceProxyHelper createWebServiceProxyHelper() {
         return new WebServiceProxyHelper();
     }
 
+    @Override
     public MCCIIN000002UV01 processPatientDiscoveryAsyncReq(PRPAIN201305UV02 message, AssertionType assertion,
-            NhinTargetCommunitiesType targets) {
+        NhinTargetCommunitiesType targets, Properties webContextProperties) {
         LOG.debug("Begin processPatientDiscoveryAsyncReq");
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/deferred/request/proxy/EntityPatientDiscoveryDeferredRequestProxyWebServiceSecuredImpl.java:61]: same comment as above for all the Entity Outbound classes.

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        LogEventRequestType auditResponse = transforms.transformResponseToAuditMsg(
+            TestPatientDiscoveryMessageHelper.createPRPAIN201305UV02Request("Gallow", "Younger", "M", "01-12-2967",
+                "1.1", "D123401", "2.2", "abd3453dcd24wkkks545"), new MCCIIN000002UV01(), assertion, createNhinTarget(),
+            NhincConstants.AUDIT_LOG_INBOUND_DIRECTION, NhincConstants.AUDIT_LOG_ENTITY_INTERFACE, Boolean.FALSE,
+            webContextProperties, NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME);
+
+        testGetEventIdentificationType(auditResponse, NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME,
+            Boolean.FALSE);
+        testGetActiveParticipantSource(auditResponse, Boolean.TRUE, webContextProperties, remoteIp);
+        testGetActiveParticipantDestination(auditResponse, Boolean.TRUE, webContextProperties, wsRequestUrl);
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java]: For Response auditing the isRequesting Boolean value should be false.

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/PassthroughInboundPatientDiscoveryDeferredRequestTest.java [hunk 5]
```diff
@@ -36,14 +36,20 @@
 import static org.mockito.Mockito.never;
 import static org.mockito.Matchers.any;
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.patientdiscovery.PatientDiscoveryAuditLogger;
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/PassthroughInboundPatientDiscoveryDeferredRequestTest.java]: This import is not used. There are several unused imports in this class.

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/StandardInboundPatientDiscoveryDeferredRequestTest.java [hunk 6]
```diff
@@ -79,36 +82,31 @@ public void standardInboundPatientDiscoveryRequest() {
         MCCIIN000002UV01 expectedResponse = new MCCIIN000002UV01();
 
         PatientDiscoveryPolicyChecker policyChecker = mock(PatientDiscoveryPolicyChecker.class);
-        AdapterPatientDiscoveryDeferredReqErrorProxyObjectFactory proxyErrorFactory = mock(AdapterPatientDiscoveryDeferredReqErrorProxyObjectFactory.class);
-        PatientDiscoveryAuditLogger auditLogger = mock(PatientDiscoveryAuditLogger.class);
-        AdapterPatientDiscoveryDeferredReqProxyObjectFactory adapterFactory = mock(AdapterPatientDiscoveryDeferredReqProxyObjectFactory.class);
+        AdapterPatientDiscoveryDeferredReqErrorProxyObjectFactory proxyErrorFactory
+            = mock(AdapterPatientDiscoveryDeferredReqErrorProxyObjectFactory.class);
+        PatientDiscoveryDeferredRequestAuditLogger auditLogger = mock(PatientDiscoveryDeferredRequestAuditLogger.class);
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/StandardInboundPatientDiscoveryDeferredRequestTest.java]: PatientDiscoveryDeferredRequestAuditLogger is used in several methods. Can it final object when the class get loaded?

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java [hunk 7]
```diff
@@ -49,6 +51,7 @@
 import gov.hhs.fha.nhinc.patientdiscovery.PatientDiscoveryPolicyChecker;
 import gov.hhs.fha.nhinc.patientdiscovery.aspect.MCCIIN000002UV01EventDescriptionBuilder;
 import gov.hhs.fha.nhinc.patientdiscovery.aspect.PRPAIN201305UV02EventDescriptionBuilder;
+import gov.hhs.fha.nhinc.patientdiscovery.audit.PatientDiscoveryDeferredRequestAuditLogger;
 import gov.hhs.fha.nhinc.patientdiscovery.entity.deferred.request.OutboundPatientDiscoveryDeferredRequestDelegate;
 import gov.hhs.fha.nhinc.patientdiscovery.entity.deferred.request.OutboundPatientDiscoveryDeferredRequestOrchestratable;
 import gov.hhs.fha.nhinc.transform.subdisc.HL7AckTransforms;
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java:54]: There are unused import in this class. Can you remove it from here?

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java [hunk 8]
```diff
@@ -137,13 +145,15 @@ public void invoke() throws ConnectionManagerException {
         OutboundPatientDiscoveryDeferredRequestDelegate delegate = mock(OutboundPatientDiscoveryDeferredRequestDelegate.class);
         PDDeferredCorrelationDao correlationDao = mock(PDDeferredCorrelationDao.class);
         ConnectionManagerCache connectionManager = mock(ConnectionManagerCache.class);
-        PatientDiscoveryAuditor auditLogger = mock(PatientDiscoveryAuditor.class);
-        OutboundPatientDiscoveryDeferredRequestOrchestratable orchestratableResponse = mock(OutboundPatientDiscoveryDeferredRequestOrchestratable.class);
-
+        PatientDiscoveryDeferredRequestAuditLogger auditLogger = mock(PatientDiscoveryDeferredRequestAuditLogger.class);
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java]: PatientDiscoveryDeferredRequestAuditLogger is used in several methods. Can it be declared as final object when class get loaded?

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java (hunk 4 above)
Comment (TabassumJafri) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java]: Agreed. it should be Boolean.False. However, AuditTransformsTest class was fixed in the recent PR and the code is not merged. Once it is merged I can fix the testcase transformResponseToAuditMsg() in PatientDiscoveryDeferredAuditTransformsTest.

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java [hunk 9]
```diff
@@ -198,36 +206,38 @@ public void noTargets() throws ConnectionManagerException {
...
         // Verify audits
-        verify(auditLogger).auditEntityDeferred201305(any(RespondingGatewayPRPAIN201305UV02RequestType.class),
-                eq(assertion), eq(NhincConstants.AUDIT_LOG_INBOUND_DIRECTION), any(String.class));
+        verify(auditLogger).auditRequestMessage(eq(request), eq(assertion), any(NhinTargetSystemType.class),
+            eq(NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION), eq(NhincConstants.AUDIT_LOG_NHIN_INTERFACE),
+            eq(Boolean.TRUE), eq(webContextProperties), eq(NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME));
 
-        verify(auditLogger).auditAck(eq(errorResponse), eq(assertion), eq(NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION),
-                eq(NhincConstants.AUDIT_LOG_ENTITY_INTERFACE));
+        /*verify(auditLogger).auditAck(eq(errorResponse), eq(assertion), eq(NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION),
```
Comment (vdmehta06) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java]: Does this comment purposely added for future use or it needs to be removed?

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+        final String localIp = "10.10.10.10";
+        final String remoteIp = "16.14.13.12";
+        final String remoteObjectUrl = "http://" + remoteIp + ":9090/source/AuditService";
+        final String wsRequestUrl = "http://" + remoteIp + ":9090/AuditService";
+
+        Properties webContextProperties = new Properties();
+        webContextProperties.setProperty(NhincConstants.WEB_SERVICE_REQUEST_URL, wsRequestUrl);
+        webContextProperties.setProperty(NhincConstants.REMOTE_HOST_ADDRESS, remoteIp);
+        webContextProperties.setProperty(NhincConstants.INBOUND_REPLY_TO,
+            "http://www.w3.org/2005/08/addressing/anonymous");
```
Comment (vdmehta06) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java]: Can we use NhincConstants.WSA_REPLY_TO as there is already a constant defined for this URL?

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java [hunk 11]
```diff
@@ -238,15 +248,17 @@ public void policyFailed() throws ConnectionManagerException {
...
-        OutboundPatientDiscoveryDeferredRequestDelegate delegate = mock(OutboundPatientDiscoveryDeferredRequestDelegate.class);
+        OutboundPatientDiscoveryDeferredRequestDelegate delegate
+            = mock(OutboundPatientDiscoveryDeferredRequestDelegate.class);
         PDDeferredCorrelationDao correlationDao = mock(PDDeferredCorrelationDao.class);
         ConnectionManagerCache connectionManager = mock(ConnectionManagerCache.class);
-        PatientDiscoveryAuditor auditLogger = mock(PatientDiscoveryAuditor.class);
-
+        PatientDiscoveryDeferredRequestAuditLogger auditLogger = mock(PatientDiscoveryDeferredRequestAuditLogger.class);
+        Properties webContextProperties = new Properties();
+        //NhinTargetSystemType target = createNhinTargetSystemType(RECEIVER_HCID);
```
Comment (vdmehta06) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java]: remove this commented code?

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/nhinclib/NullChecker.java [hunk 12]
```diff
@@ -59,7 +59,7 @@ public static boolean isNotNullish(Collection<?> value) {
         return (!isNullish(value));
     }
 
-    public static boolean isNullishIgnoreSpace(String value) {
+    public static boolean isNotNullishIgnoreSpace(String value) {
         return (value != null && !value.trim().isEmpty());
```
Comment (christophermay07) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/nhinclib/NullChecker.java]: Unnecessary parens around return value.

File: Product/Production/Gateway/PatientDiscovery_10/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/_10/gateway/ws/EntityPatientDiscoveryDeferredRequestSecured.java (hunk 1 above)
Comment (TabassumJafri) [inline Product/Production/Gateway/PatientDiscovery_10/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/_10/gateway/ws/EntityPatientDiscoveryDeferredRequestSecured.java:68]: I will remove the WebContextProperties for Outbound PD deferred classed

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransforms.java [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+        return auditMsg;
+    }
+
+    private AuditMessageType getQueryParamsParticipantObjectIdentificationForResponse(PRPAIN201305UV02 request,
+        AuditMessageType auditMsg) throws JAXBException {
+
+        ParticipantObjectIdentificationType participantObject = buildBaseParticipantObjectIdentificationType(
+            PRPAIN201305UV02Parser.getQueryId(request));
+        // For PDDeffered request we are populating queryByParameter from request as we don't have queryByParameter
```
Comment (christophermay07) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransforms.java]: Should read "Deferred" in comment

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/deferred/request/proxy/EntityPatientDiscoveryDeferredRequestProxyWebServiceSecuredImpl.java [hunk 14]
```diff
@@ -55,23 +56,26 @@ protected WebServiceProxyHelper createWebServiceProxyHelper() {
...
+            RespondingGatewayPRPAIN201305UV02SecuredRequestType securedRequest
+                = new RespondingGatewayPRPAIN201305UV02SecuredRequestType();
             securedRequest.setNhinTargetCommunities(targets);
             securedRequest.setPRPAIN201305UV02(message);
 
             response = (MCCIIN000002UV01) client.invokePort(EntityPatientDiscoverySecuredAsyncReqPortType.class,
-                    "processPatientDiscoveryAsyncReq", securedRequest);
+                "processPatientDiscoveryAsyncReq", securedRequest);
         } catch (Exception ex) {
             LOG.error("Error calling processPatientDiscoveryAsyncReq: " + ex.getMessage(), ex);
```
Comment (christophermay07) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/deferred/request/proxy/EntityPatientDiscoveryDeferredRequestProxyWebServiceSecuredImpl.java]: When you're in the code, if you see ex.getMessage() in a file you're already touching, can you change to ex.getLocalizedMessage()?  Little improvements add up over time :)

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+
+        testGetEventIdentificationType(auditResponse, NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME,
+            Boolean.FALSE);
+        testGetActiveParticipantSource(auditResponse, Boolean.FALSE, webContextProperties, remoteIp);
+        testGetActiveParticipantDestination(auditResponse, Boolean.FALSE, webContextProperties, wsRequestUrl);
+        assertNotNull("ParticipantObjectIdentification is null", auditResponse.getAuditMessage().
+            getParticipantObjectIdentification());
+        assertFalse("ParticipantObjectIdentification is empty", auditResponse.getAuditMessage().
+            getParticipantObjectIdentification().isEmpty());
+        assertEquals("ParticipantObjectIdentification size is 1", auditResponse.getAuditMessage().
```
Comment (christophermay07) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java]: This message is wrong; I would suggest something along the lines of "ParticipantObjectIdentification should only have 1 entry" (at this point, we know that there is at least 1 entry because we already checked "isEmpty()"), but pick a message you're comfortable with.

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/StandardInboundPatientDiscoveryDeferredRequestTest.java [hunk 16]
```diff
@@ -59,11 +62,13 @@
  */
 public class StandardInboundPatientDiscoveryDeferredRequestTest {
 
+    private final PatientDiscoveryDeferredRequestAuditLogger auditLogger = mock(PatientDiscoveryDeferredRequestAuditLogger.class);
```
Comment (christophermay07) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/StandardInboundPatientDiscoveryDeferredRequestTest.java]: If you need to make another commit, can you wrap this line?  (Don't make a commit just for this)

**Comment by christophermay07** [comment]
I've reviewed this PR, endorsing for merge without further comment.

**Comment by vdmehta06** [comment]
Reviewed and good for merge

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/AbstractInboundPatientDiscoveryDeferredRequest.java [hunk 17]
```diff
@@ -27,10 +27,12 @@
 package gov.hhs.fha.nhinc.patientdiscovery.inbound.deferred.request;
 
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/inbound/deferred/request/AbstractInboundPatientDiscoveryDeferredRequest.java]: This import is unused. It can be taken care of when you do another commit.

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java [hunk 18]
```diff
@@ -79,6 +83,7 @@
     private MCCIIN000002UV01 expectedResponse;
     private II patientId;
     private AssertionType assertion;
+    private final PatientDiscoveryDeferredRequestAuditLogger auditLogger = mock(PatientDiscoveryDeferredRequestAuditLogger.class);
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/StandardOutboundPatientDiscoverDeferredRequestTest.java]: Line wrap.

File: Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/PassthroughOutboundPatientDiscoveryDeferredRequestTest.java [hunk 19]
```diff
@@ -56,29 +58,29 @@ public void invoke() {
         AssertionType assertion = new AssertionType();
         NhinTargetCommunitiesType targets = new NhinTargetCommunitiesType();
         MCCIIN000002UV01 expectedResponse = new MCCIIN000002UV01();
-
```
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/outbound/deferred/request/PassthroughOutboundPatientDiscoveryDeferredRequestTest.java:59]: webContextProperties below here is not used.
Properties webContextProperties = new Properties();

**Comment by alameluchidambaram** [comment]
I have reviewed this PR and good with merge. The comments for cosmetic changes can be handled now or later and that's fine with me.

**Comment by alameluchidambaram** [comment]
Reviewed PR and going to merge when PR validator is done.
